main.js

```diff
@@ -0,0 +1,430 @@
+        winCondition(rowToPlacePiece, column_clicked, gameArray, playerNumber);
+        changeTurn();
+        console.log('Piece placement function working');
+        console.log('This is: ', this);
+    }
+
+    function columnIndex(jQueryObj) {
+
+        column_clicked = jQueryObj.attr("id");
+
+        console.log('The column clicked was: ', column_clicked);
+    }
+
+
+    function rowIndex() {
+        for (i = 6; i >= 0; i--) {
+            if (gameArray[i][column_clicked].token < 1) {
+                rowToPlacePiece = i;
+                return rowToPlacePiece;
+            }
+        }
+    }
+
+    function rowForPiece() {
+        putPieceInRow = piecesInColumn - 1;
+        return putPieceInRow;
+    };
+
+
+// 2x2 SQUARE BONUS
+    function squareBonusCheck(boardArrayClick) {
+        if (boardArray[row - 1][column + 1] === 'X') {
+            if (boardArray[row - 0][column + 1] === 'X' && boardArray[row - 1][column - 0] === 'X') {
+                console.log('SQUAREBONUS top right');
+                return;
+            }
+            ;
+        }
+        ;
+        if (boardArray[row - 1][column - 1] === 'X') {
+            if (boardArray[row - 0][column - 1] === 'X' && boardArray[row - 1][column - 0] === 'X') {
+                console.log('SQUAREBONUS top left');
+                return;
+            }
+            ;
+        }
+        ;
+        if (boardArray[row + 1][column - 1] === 'X') {
+            if (boardArray[row - 0][column - 1] === 'X' && boardArray[row + 1][column - 0] === 'X') {
+                console.log('SQUAREBONUS bottom left');
+                return;
+            }
+            ;
+        }
+        ;
+        if (boardArray[row + 1][column + 1] === 'X') {
+            if (boardArray[row - 0][column + 1] === 'X' && boardArray[row + 1][column - 0] === 'X') {
+                console.log('SQUAREBONUS bottom right');
+                return;
+            }
+            ;
+        }
+        ;
+    };
+
+    function winCondition(row, column, array, player) {
+        // console.log('chickens');
+        // console.log('row', row);
+        // console.log('column', column)
+        // console.log('array', array)
+        // console.log('player:', player);
+        row = Number(row);
+        column = Number(column);
+        if (array[row][column].token !== player) {
+            return false
+        }
+        var row_direction_array = [0, -1, -1, -1];
+        var column_direction_array = [-1, 1, 0, -1];
+
+        for (var x = 0; x < row_direction_array.length; x++) {
+            var row_direction = row_direction_array[x];
+            var column_direction = column_direction_array[x];
+
+            var check_row = row + row_direction;
+            var check_column = column + column_direction;
+            var inline_counter = 1;
+
+            for (var i = 0; i < 2; i++) {
+                if (array[check_row] === undefined) {
+                }
```

Comment:
having empty if statements like this will be highly frowned upon

You would be better doing it as the inverse
```
if(array[check_row] !== undefined && array[check_row][check_column] !== undefined){
```